src/comm/MockLink.cc

```diff
@@ -1052,6 +1055,53 @@ void MockLink::_handleFTP(const mavlink_message_t& msg)
     _mockLinkFTP->mavlinkMessageReceived(msg);
 }
 
+void MockLink::_handleInProgressCommandLong(const mavlink_command_long_t& request)
+{
+    uint8_t commandResult = MAV_RESULT_UNSUPPORTED;
+
+    switch (request.command) {
+    case MockLink::MAV_CMD_MOCKLINK_RESULT_IN_PROGRESS_ACCEPTED:
+        // Test command which sends in progress messages and then acceptance ack
+        commandResult = MAV_RESULT_ACCEPTED;
+        break;
+    case MockLink::MAV_CMD_MOCKLINK_RESULT_IN_PROGRESS_FAILED:
+        // Test command which sends in progress messages and then failure ack
+        commandResult = MAV_RESULT_FAILED;
+        break;
+    case MockLink::MAV_CMD_MOCKLINK_RESULT_IN_PROGRESS_NO_ACK:
+        // Test command which sends in progress messages and then never sends final result ack
```

Comment:
I assume this is to test legacy behavior? It should not be encouraged going forward.

Reply:
It's to simulate a noisy comm link where the final ack is lost on the wire. QGC handles that case an reports an error back to the handler.